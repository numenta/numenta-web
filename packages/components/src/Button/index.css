/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._disabled:disabled {
  background-color: #999 !important;
}


.default {
  composes: ba from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: ph4 from 'tachyons';
  composes: pv3 from 'tachyons';
  background-color: transparent;
  color: #1aa0db;
}

.light {
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: pa1 from 'tachyons';
  background-color: transparent;
  color: #1aa0db;
}

.primary {
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: ph4 from 'tachyons';
  composes: pv3 from 'tachyons';
  composes: white from 'tachyons';
  background-color: #1aa0db;
}

.short {
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: h2 from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: pv0 from 'tachyons';
  composes: relative from 'tachyons';
  composes: white from 'tachyons';
  background-color: #1aa0db;
  top: -0.1rem;
}


.button {
  composes: br2 from 'tachyons';
  composes: dim from 'tachyons';
  composes: flex-auto from 'tachyons';
  composes: ma0 from 'tachyons';
  composes: pointer from 'tachyons';
  composes: _disabled;
}
