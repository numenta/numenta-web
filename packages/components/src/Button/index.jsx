// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const Button = ({children, disabled, onClick, theme, type}) => {
  const classes = [styles.button, styles[theme]]

  if (disabled) classes.push(styles.disabled)

  return (
    <button
      className={classNames(...classes)}
      disabled={disabled}
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  )
}

Button.propTypes = {
  children: React.PropTypes.node.isRequired,
  disabled: React.PropTypes.bool,
  onClick: React.PropTypes.func,
  theme: React.PropTypes.oneOf(['default', 'light', 'primary', 'short']),
  type: React.PropTypes.oneOf(['button', 'menu', 'reset', 'submit']),
}

Button.defaultProps = {
  disabled: false,
  theme: 'default',
  type: 'button',
}

export default Button
