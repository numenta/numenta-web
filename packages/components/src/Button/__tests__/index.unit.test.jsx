import React from 'react'
import renderer from 'react-test-renderer'

import Button from '../../Button'


describe('Button React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <Button
        onClick={() => {}}
      >
        Title
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Disabled correctly', () => {
    const component = renderer.create(
      <Button
        disabled={true}
        onClick={() => {}}
      >
        Title
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Type:Submit correctly', () => {
    const component = renderer.create(
      <Button
        onClick={() => {}}
        type="submit"
      >
        Title
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Theme:Light correctly', () => {
    const component = renderer.create(
      <Button
        onClick={() => {}}
        theme="light"
      >
        Title
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Theme:Primary correctly', () => {
    const component = renderer.create(
      <Button
        onClick={() => {}}
        theme="primary"
      >
        Title
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Theme:Short correctly', () => {
    const component = renderer.create(
      <Button
        onClick={() => {}}
        theme="short"
      >
        Title
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
