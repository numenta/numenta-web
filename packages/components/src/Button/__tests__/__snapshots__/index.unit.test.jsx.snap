exports[`Button React component Renders Default correctly 1`] = `
<button
  className="button default"
  disabled={false}
  onClick={[Function]}
  type="button">
  Title
</button>
`;

exports[`Button React component Renders Disabled correctly 1`] = `
<button
  className="button default disabled"
  disabled={true}
  onClick={[Function]}
  type="button">
  Title
</button>
`;

exports[`Button React component Renders Theme:Light correctly 1`] = `
<button
  className="button light"
  disabled={false}
  onClick={[Function]}
  type="button">
  Title
</button>
`;

exports[`Button React component Renders Theme:Primary correctly 1`] = `
<button
  className="button primary"
  disabled={false}
  onClick={[Function]}
  type="button">
  Title
</button>
`;

exports[`Button React component Renders Theme:Short correctly 1`] = `
<button
  className="button short"
  disabled={false}
  onClick={[Function]}
  type="button">
  Title
</button>
`;

exports[`Button React component Renders Type:Submit correctly 1`] = `
<button
  className="button default"
  disabled={false}
  onClick={[Function]}
  type="submit">
  Title
</button>
`;
